testdata/project-v3-v1beta1/go.mod

```diff
@@ -3,8 +3,8 @@ module sigs.k8s.io/kubebuilder/testdata/project-v3-v1beta1
 go 1.17
 
 require (
-	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.13.0
+	github.com/onsi/ginkgo/v2 v2.1.3
+	github.com/onsi/gomega v1.17.0
```

Comment:
This version is probably incompatible. We need to revert the dep if/when the v1beta1 be used see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/commons.go#L38

Reply:
@camilamacedo86 I might be totally missing something, but what does the ginkgo version have to do with the CRD version? I would expect ginkgo/v2 to work with projects that use v1beta1 CRDs. Is it just that the other dependencies we're required to use with v1beta1 CRDs use ginkgo/v1, so we're concerned that there would be issues if the kubebuilder project used v2, but the project's dependencies used v1?

The one important thing to cover is this:
- Regardless of the CRD version (or anything else), if a project was originally scaffolded with ginkgo v1, we should continue scaffolding new APIs and webhooks with ginkgo v1.

I'm not sure I saw anything for that in my review, but if we have that covered, disregard.

Reply:
When we scaffold CRD v1beta1 we need to ensure that we use compatible deps with controller-gen v0.6.2 and controller-runtime v0.9.2 since these are the latest versions where we can use v1beta1 with. So, my point is that:
- We need to check if the ginkgo v2 is compatible with or not, if not we need to try to downgrade ([here](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/commons.go#L38))
- However, we have changes in the scaffolds as well, so that makes it very complicated if the ginkgo v2 be incompatible. :-( 
- We will also need to ensure that. the changes in the scaffolds are not a breaking change for projects which were already built with old KB versions and their previous versions. 

Since we need to change the scaffold because of the deprecations it seems like a breaking change that we would only be able to address in the go/v4 plugin. After checking the migration steps I am thinking that we should track it to address with go/v4, see: https://github.com/kubernetes-sigs/kubebuilder/milestone/26

@joelanford wdyt? 

/hold 

(it shows a breaking change and cannot be addressed on go/v3)